session/txnmanager.go

```diff
@@ -139,6 +139,8 @@ func (m *txnManager) newProviderWithRequest(r *sessiontxn.EnterNewTxnRequest) se
 		switch m.sctx.GetSessionVars().IsolationLevelForNewTxn() {
 		case ast.ReadCommitted:
 			return isolation.NewPessimisticRCTxnContextProvider(m.sctx, r.CausalConsistencyOnly)
+		case ast.RepeatableRead:
```

Comment:
Though we do not support READ-UNCOMMITTED , it is still allowed by tidb and will use the RR isolation. So I think we can do it like this:

```
		switch m.sctx.GetSessionVars().IsolationLevelForNewTxn() {
		case ast.ReadCommitted:
			return isolation.NewPessimisticRCTxnContextProvider(m.sctx, r.CausalConsistencyOnly)
        case ast.Serializable:
             // TODO: 
		case ast.RepeatableRead:
            return isolation.NewPessimisticRRTxnContextProvider(m.sctx, r.CausalConsistencyOnly)
        }
```

Reply:
Good suggestion, I have updated it.